### DISCUSSION TITLE
keycloak + zabbix with HTTP AUTH

### DISCUSSION THREAD

**Discussion by kv-open** [OpenIDC/mod_auth_openidc#601] [General]
Hello!

I want to configure keycloak AUTH to modern zabbix version 5.2
First I configured simple test web-page to be sure that "KC" works correctly

my apache  config is
```
<VirtualHost *:80>

    ServerAdmin webmaster@localhost
    DocumentRoot /var/www/html

    #this is required by mod_auth_openidc
	OIDCProviderMetadataURL https://KEYCLOAK_SERVER_IP/auth/realms/zabbix/.well-known/openid-configuration
	OIDCRedirectURI http://APACHE_SERVER_IP/protected_by_oidc_rp.php/callback
	OIDCCryptoPassphrase 0123456789
	OIDCClientID zabbix
	OIDCClientSecret 9ac67ff0-5733-45ea-b910-2b8dcd1386d2
	# See https://github.com/Reposoft/openidc-keycloak-test/issues/7
	#OIDCProviderTokenEndpointAuth client_secret_basic
    OIDCSSLValidateServer Off
	OIDCRemoteUserClaim email
	OIDCScope "openid email"

    <Location "/protected_by_oidc_rp.php">
        AuthType openid-connect
        Require valid-user
    </Location>
</VirtualHost>
```
and everything is ok.
After authorization in KC I get my page

Second
I Enabled HTTP authentication in zabbix - it's just simple basic auth

i reconfigured my apache location

```
<VirtualHost *:80>
......
......
......
    <Location "/">
        AuthType openid-connect
        Require valid-user
        ProxyPreserveHost On
        ProxyPass http://ZABBIX_SERVER_IP/
        ProxyPassReverse http://ZABBIX_SERVER_IP/
    </Location>
</VirtualHost>
```
And last thing that i have to do is add header 

    <Location "/">
         ...
         ...
        RequestHeader set Authorization "Basic <SOME_VARRIABLE>"
    </Location>
</VirtualHost>

And my question is 
How i can get username from keycloak varriable e.g. "preferred_username" and convert it to base64?

**Comment by zandbelt** [comment]
I *think* you may be able to use something like described in: https://serverfault.com/questions/215537/apache-rewriterule-with-a-rewritemap-variable-substitution-for-the-val-argument
